projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-date/gov-uk-date.component.ts

```diff
@@ -1,24 +1,54 @@
-import {Component, Input, OnInit} from '@angular/core';
+import {
+  Component,
+  Input,
+  OnInit
+} from '@angular/core';
+import { ValidationErrors, ValidatorFn } from '@angular/forms';
 import { ErrorMessagesModel } from '../../models/error-messages-model';
 import { GovUiConfigModel } from '../../models/gov-ui-config-model';
 /*
-* Gov UK Date Component
-* Responsible for displaying 3 input fields:
-* day / month / year
-* displaying errorMessage messages
-* */
+ * Gov UK Date Component
+ * Responsible for displaying 3 input fields:
+ * day / month / year
+ * displaying errorMessage messages
+ * */
 @Component({
   selector: 'xuilib-gov-uk-date',
   templateUrl: './gov-uk-date.component.html',
   styleUrls: ['./gov-uk-date.component.scss']
 })
 export class GovUkDateComponent implements OnInit {
-  constructor() { }
+  constructor() {}
   @Input() public config: GovUiConfigModel;
-  // { id: string };
   @Input() public errorMessage: ErrorMessagesModel;
   @Input() public formGroup: any;
 
+  public month: any;
+  public day: any;
+  public year: any;
+
   public ngOnInit(): void {
+    this.day = `${this.config.id}_day`;
+    this.month = `${this.config.id}_month`;
+    this.year = `${this.config.id}_year`;
+    const dateValidator = this.DateValidator();
+    this.formGroup.get(this.day).setValidators(dateValidator);
+  }
+
+  private isValidDate(d: any, month: number): boolean {
+    const dateCheck = d instanceof Date && !isNaN(d.getTime());
+    const leapYearCheck = d.getMonth() === month;
```

Comment:
Discussed with Tarik regarding why don't we use moment, however as the package is not installed in the common lib it was deemed VanillaJS is suffice in this instance.